File: 45-jump-game-ii/45-jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        dp = [float("inf") for _ in range(len(nums))]
        dp[0] = 0
        for index , num in enumerate(nums):
            for j in range(index+1,index + num + 1):
                if j < len(nums):
                    dp[j] = min(dp[j],dp[index] + 1)
            # print(dp)
        return dp[-1]
            

File: 45-jump-game-ii/45-jump-game-ii.java
class Solution {
    public int jump(int[] nums) {
        int[] dp = new int[nums.length];
        Arrays.fill(dp,Integer.MAX_VALUE);
        dp[0] = 0;
        for (int curr = 0 ; curr < nums.length; curr ++){
            for (int i = curr + 1 ; i < curr + nums[curr] + 1;i ++){
                if (i < nums.length)
                    dp[i] = Math.min(dp[i],dp[curr] + 1);
                
            }
        }
        
        return dp[nums.length - 1];
    }
}